lib/internal/streams/duplex.js

```diff
@@ -63,6 +63,24 @@ function Duplex(options) {
   if (!(this instanceof Duplex))
     return new Duplex(options);
 
+  this._events ??= {
```

Comment:
What's the performance benefit/impact if you convert each one of them to an array? (Same for the rest)

Reply:
Pretty bad. Usually there is only one listener.